### ISSUE TITLE
Comments on #define are not highlighted

### ISSUE THREAD

**Issue by cjastone** [atom/language-c#299]
### Prerequisites

* [ ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Comments on a line that starts with #define are not highlighted

### Steps to Reproduce

1. Create a line of code with `#define` (e.g. `#define MY_VAR 123') and add a comment to the end (either `// comment` or `/* commend */`
2. Observe that comment is not highlighted correctly.

**Expected behavior:** Comment is dark grey and italicised.

**Actual behavior:** Comment remains default light grey and non-italicised.

**Reproduces how often:** Consistently for the above example.

### Versions

Atom    : 1.32.0
Electron: 2.0.9
Chrome  : 61.0.3163.100
Node    : 8.9.3
Windows 10 x64 1803

**Comment by someburner** [comment]
I commented in [this](https://github.com/atom/atom/issues/18327) issue that in 1.32.0 syntax highlighting is broken when `Use Tree Sitter Parsers` is enabled. What I'm seeing is similar to what's described by OP in that issue, but for C what happens is highlighting gets messed up when `#ifdef`s are present.

Not sure if this is related to this issue, or is a new issue, so commenting here first. I'm guessing it might be related since, for example, if you do:

```c
#if 0
void someCodeHere(void);
#endif
```

In 1.31.2 and earlier that code will be grayed-out. But in 1.32.0 it is just black text when `Use Tree Sitter Parsers` is ticked. When unticked it looks as it should.

**Comment by maxbrunsfeld** [comment]
@someburner Yeah, that is a known change due to our new parsing system. It is possible for us to fix, but probably not a super high priority, because it was never a very general behavior. The old parser was just hard-coded to look for `#if 0`, (as opposed to actually determining if the macro evaluated to false), which would be much more generally useful.

For the time being, if that behavior is very useful to you, you can keep it by unchecking the `Use Tree-sitter Parsers` checkbox, but we are moving away from maintaining the old parsers, because the new parsers fix a large number of issues. For example, we can finally highlight types and struct fields consistently, which we never could do before.

**Comment by maxbrunsfeld** [comment]
You can also fold `#ifdef` blocks now, which you usually couldn't do before because the contents are not traditionally indented.

**Comment by someburner** [comment]
Okay maybe my issue is different and/or related to the syntax I'm using ([fizzy](https://github.com/jglovier/fizzy)).

With Treesitter ticked, I get the following two issues:

### 1: built-in types not highlighted correctly (int, bool, etc)

### 2: Multi-line macros do not highlight

Example:

![image](https://user-images.githubusercontent.com/9430255/47751661-a98e4300-dc4f-11e8-9116-62492ae5ebbe.png)

Vs unticked:

![image](https://user-images.githubusercontent.com/9430255/47751761-e65a3a00-dc4f-11e8-8d41-eb948f28f7d3.png)

I can make a new issue if needed.

**Comment by Ben3eeE** [comment]
@someburner Please make a new issue and include the code in text so we can take a look, thanks 🙇

**Comment by maxbrunsfeld** [comment]
What theme are you using? Can you compare with the built in ‘one light syntax’ theme? The highlighting of types used to only work for built in types, but now should work for all types.

**Comment by someburner** [comment]
I'm using this one: https://github.com/jglovier/fizzy

But these issues are also present when using standard themes like "Atom Light".

A good example to see the behavior is: https://github.com/nodejs/http-parser/blob/master/http_parser.h

**Comment by maxbrunsfeld** [comment]
:+1: Yeah, agreed. When I said that issue was "already fixed", I mean that the bug has been fixed, but the fix has not yet been published. We're working on getting Atom 1.32.1 out right now.

**Comment by someburner** [comment]
Ah, okay thanks. I'll wait for 1.32.1 and see how it goes.

**Comment by maxbrunsfeld** [comment]
But actually, the issue that's fixed is just the macro body thing. I *am* still curious about any problems you're seeing with types being highlighted, because as far as I know, that should work on 1.32. Does *that* problem show up for you on the default theme?

**Comment by someburner** [comment]
Yeah actually only with fizzy. But when Treesitters is unticked it does highlight. Are there some new directives needed to get that working? I'm a CSS noob :)

**Comment by maxbrunsfeld** [comment]
Hmm, I'll take a look; our intention was to maintain compatibility with most themes in general use, so sorry about that. Maybe there's some extra classes we can add. Unfortunately, that work won't make it into 1.32.1, but it's on our radar.

**Comment by cjastone** [comment]
Thanks @maxbrunsfeld - have the update and this is back to normal, keep up the great work everyone!

**Comment by Ben3eeE** [comment]
Seems like this is because some of the primitive types are scoped as `support.support.type` instead of `support.storage.type`. I opened a PR.
